File: DictionaryAttack.rb
#!/usr/bin/ruby
#ddrage0 Ruby Attack Week6.5 24/02/2022

require 'digest'
#read the password

passwordHash = "5994471abb01112afcc18159f6cc74b4f511b99806da59b3caf5a9c173cacfc5"

#open the wordlist and repeat for each word

IO.foreach("wordlist.txt") do | word |
    
    word.chomp!

    #hash the word

    wordlistHash = Digest::SHA256.hexdigest(word)

    puts("Trying password #{word}: #{wordlistHash}")

    #if the hash is the same as the correct password's hash then we have cracked the password!

    if(wordlistHash == passwordHash)
        puts("Password has been cracked! it was #{word}")

        return
    
    end

end

File: DictionaryAttack.pl
#!/usr/bin/perl
#ddrage0 Perl Attack Week6.5

use Digest::SHA qw(sha256_hex);

#read the password hash

$passwordHash = "04e77bf8f95cb3e1a36a59d1e93857c411930db646b46c218a0352e432023cf2";

#open the wordlist

open(WORDLIST,"<wordlist.txt") or die "Couldn't open wordlist, $!";

#repeat for each word

while(<WORDLIST>){
    chomp;
    $word = $_;

    #hash the word
    $wordlistHash = sha256_hex($word);

    print "$word: $wordlistHash\n";

    #if the hash is the same as the correct passowrd's hash then we have cracked the password!
    if($wordHash eq $passwordHash) {
        print("Password has been cracked! it was $word\n");

        exit;
    }

}